src/generators.ts

```diff
@@ -8,185 +8,173 @@ export interface GeneratorOptions {
+
   const firstFrame = simplifiedFrames[0]
   if (!firstFrame)
     throw new Error('The document contains no frames.')
 
-  writer.block(() => {
-    const machineId = normalizeString(pageName)
-
-    // Machine id
-    writer.write('id:').space().quote().write(machineId).quote().write(',').newLine()
-
-    // Initial State
-    writer
-      .write('initial:')
-      .space()
-      .quote()
-      .write(normalizeString(firstFrame.name))
-      .quote()
-      .write(',')
-      .newLine()
+  const machineId = normalizeString(pageName)
 
-    // Machine states
-    writer.write('states:').block(() => {
+  w.stateMachineConfig(() => {
+    w.stateId(machineId)
+    w.initialState(firstFrame.name)
+    w.statesBlock(() => {
       for (const simplifiedFrame of simplifiedFrames) {
-        // TODO: Support fragments with same name
+        // TODO: Support frames with same name
         const frameStateId = normalizeString(simplifiedFrame.name)
 
-        // State name
-        writer
-          .write(frameStateId)
-          .write(':')
-          .inlineBlock(() => {
-            // State body
-
-            const childNodesThatNavigate = interactiveNodes.filter(
-              // TODO: support nested nodes
-              element => element.parentFrameId === simplifiedFrame.id,
-            )
-
-            const noMachineEvents = childNodesThatNavigate.length === 0
-            if (noMachineEvents) {
-              writer.writeLine('type: \'final\'')
-              return
-            }
-
-            // TODO: narrow down to specific types
-            const childNodesThatNavigateWithDelay = childNodesThatNavigate.filter(
-              node => 'delay' in node,
-            )
-            const childNodesThatNavigateWithoutDelay = childNodesThatNavigate.filter(
-              node => !('delay' in node),
-            )
-
-            const delayedEvents = childNodesThatNavigateWithDelay.map((childNodeThatNavigate) => {
-              const eventName = normalizeString(
-                `${
-                  childNodeThatNavigate.triggerType
-                }_${childNodeThatNavigate.generatedName.toUpperCase()}`,
-              )
-
-              const delay: number
-                // @ts-expect-error TS does not kno the proper triggerType because types are not enough narrowed down in `childNodesThatNavigateWithDelay` definition
-                = childNodeThatNavigate.delay
-
-              const destinationFrame = simplifiedFrames.find(
-                ({ id }) => childNodeThatNavigate.destinationFrameId === id,
-              )
-
-              if (!destinationFrame)
-                throw new Error(`Frame ${childNodeThatNavigate.destinationFrameId} not found`)
-
-              return {
-                delay,
-                eventName,
-                destinationFrame,
-                destinationStateName: `${eventName}_AFTER_${delay}`,
+        // ex. Frame_1
+        w.stateBlock(frameStateId, () => {
```

Comment:
Any documentation to be commented here around w.* methods?

Reply:
Like? What would you expect here? (I mean, you are right but I would like to know what you have in mind 😊)

Reply:
Methods like stateBlock are vague to me so I'd like to know what they do and how they work and why we're using them.

Reply:
Makes sense!

Reply:
Here I am, @Aslemammad!!!! Let me know if this makes sense to you 😊
https://github.com/NoriSte/figma-to-xstate/pull/11/commits/89b2024bf40efd29290995fc6efcdbe22d9cfb35

Reply:
That's so much better! It's clear to me right now. Thank you! I wish I learn this way of writing comments :)

Reply:
In a pure "show, don't tell" mode, I improved them
https://github.com/NoriSte/figma-to-xstate/pull/11/commits/183aa1ed637aa71a370b149f96602be37219a3f5
:)